examples/gno.land/r/gnoland/faucet/faucet.gno

```diff
@@ -4,65 +4,95 @@ import (
 	"errors"
 	"std"
 
+	"gno.land/p/demo/avl"
 	"gno.land/p/demo/ufmt"
 )
 
 var (
 	// configurable by admin.
-	gAdminAddr       std.Address = "g1u7y667z64x2h7vc6fmpcprgey4ck233jaww9zq"
-	gControllerAddr  std.Address = "g1u7y667z64x2h7vc6fmpcprgey4ck233jaww9zq"
-	gPerTransferSend             = std.Coins{std.Coin{"ugnot", 1000000}}
-	gInPause                     = false
-	gMessage                     = "# Community Faucet.\n\n"
+	gAdminAddr std.Address = std.Address("g1jg8mtutu9khhfwc4nxmuhcpftf0pajdhfvsqf5")
+
+	gControllers     = avl.NewMutTree()
+	gControllersSize = 10 // limit it to 10
+
+	gInPause = false
+
+	gMessage = "# Community Faucet.\n\n"
 
 	// internal vars, for stats.
 	gTotalTransferred std.Coins
 	gTotalTransfers   = uint(0)
+
+	// per request limit, 350 gnot
+	gLimit std.Coin = std.Coin{"ugnot", 350000000}
 )
 
-func Transfer(to std.Address) error {
+func Transfer(to std.Address, send int64) string {
```

Comment:
What do you think about using parametrized returns? I'm honestly not sure if this is supported yet (cc @harry-hov).

For ex:
```go
func Transfer(to std.Address, send int64) (res string) {
// ...
}
```

This way you don't need to have a hanging `return ""` at the end of each method, as `res` is `""` by default

Reply:
I tried to avoid named return parameters in a function signature. That is to prevent variable shadowing in the function implementation, even for a simple case like this. I can spend less time considering whether or not to use named return parameters.